src/main/java/org/opensearch/ad/rest/RestValidateDetectorAction.java

```diff
@@ -1,44 +1,144 @@
 package org.opensearch.ad.rest;
 
 import static org.opensearch.rest.RestRequest.Method.POST;
-import static org.opensearch.rest.RestStatus.NOT_FOUND;
-import static org.opensearch.rest.RestStatus.OK;
 
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.Collections;
 
+import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.opensearch.ad.AnomalyDetectorExtension;
+import org.opensearch.ad.common.exception.ADValidationException;
 import org.opensearch.ad.constant.CommonErrorMessages;
+import org.opensearch.ad.model.AnomalyDetector;
+import org.opensearch.ad.model.DetectorValidationIssue;
+import org.opensearch.ad.model.ValidationAspect;
+import org.opensearch.ad.settings.AnomalyDetectorSettings;
 import org.opensearch.ad.settings.EnabledSetting;
+import org.opensearch.ad.transport.ValidateAnomalyDetectorAction;
+import org.opensearch.ad.transport.ValidateAnomalyDetectorRequest;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentType;
 import org.opensearch.extensions.rest.ExtensionRestRequest;
 import org.opensearch.extensions.rest.ExtensionRestResponse;
 import org.opensearch.rest.RestHandler.Route;
 import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.rest.action.RestToXContentListener;
 import org.opensearch.sdk.ExtensionRestHandler;
+import org.opensearch.sdk.ExtensionsRunner;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ad.util.RestHandlerUtils.TYPE;
+import static org.opensearch.ad.util.RestHandlerUtils.VALIDATE;
 
 public class RestValidateDetectorAction implements ExtensionRestHandler {
     private final Logger logger = LogManager.getLogger(RestValidateDetectorAction.class);
+    private final OpenSearchClient sdkClient;
+    private final NamedXContentRegistry xContentRegistry;
+
+    public static final Set<String> ALL_VALIDATION_ASPECTS_STRS = Arrays
+            .asList(ValidationAspect.values())
+            .stream()
+            .map(aspect -> aspect.getName())
+            .collect(Collectors.toSet());
 
+    public RestValidateDetectorAction(ExtensionsRunner runner, AnomalyDetectorExtension extension){
+        this.xContentRegistry = runner.getNamedXContentRegistry().getRegistry();
+        this.sdkClient = extension.getClient();
+    }
     @Override
     public List<Route> routes() {
         return List.of(new Route(POST, "/detectors/_validate"), new Route(POST, "/detectors/_validate/{type}"));
```

Comment:
While this works, there's a better syntax available with the `BaseExtensionRestHandler`.  So you can:
1. change `implements ExtensionRestHandler` to `extends BaseExtensionRestHandler`.
2. change this method to `List<RouteHandler> routeHandlers()`
3. make the list of `new RouteHandler(POST, "/detectors/_validate", (r) -> handlePostRequest(r))` (and another one for the other route pointing to the same method).

See 
https://github.com/opensearch-project/anomaly-detection/blob/71010ca7f99c889b9d8a9a9cba3a3b2f0ba8afc9/src/main/java/org/opensearch/ad/rest/RestCreateDetectorAction.java#L63-L66

Reply:
ACk